src/foam/nanos/auth/Group.js

```diff
@@ -128,6 +143,41 @@ foam.CLASS({
 
         return false;
       }
+    },
+    {
+      name: 'getAppConfig',
+      javaReturns: 'AppConfig',
+      args: [
+        {
+          name: 'x',
+          javaType: 'X'
+        }
+      ],
+      javaCode: `
+DAO userDAO         = (DAO) x.get("localUserDAO");
+DAO groupDAO        = (DAO) x.get("groupDAO");
+AppConfig config    = (AppConfig) ((AppConfig) x.get("appConfig")).fclone();
+
+Session session = x.get(Session.class);
+if ( session != null ) {
```

Comment:
So how does this work for emails initiated by cron, when the session is null?  
I don't see any other relevant source in the PR?

Reply:
For emails initiated by cron, the session is not null, it's a session with userId 1. So for cron, it will check the system group url first,  if it's empty, then use the default appconfig url. I changed some code for ScheduledEmail.java for cron, but it's not in foam folder. I think I should create PR for foam first.

Reply:
re: ScheduledEmail.java - sure create a seperate PR if you like.